geo-traits/src/polygon.rs

```diff
@@ -0,0 +1,134 @@
+use std::marker::PhantomData;
+
+use crate::iterator::PolygonInteriorIterator;
+use crate::line_string::UnimplementedLineString;
+use crate::{Dimensions, LineStringTrait};
+use geo_types::{CoordNum, LineString, Polygon};
+
+/// A trait for accessing data from a generic Polygon.
+///
+/// A `Polygon`’s outer boundary (_exterior ring_) is represented by a
+/// [`LineString`][LineStringTrait]. It may contain zero or more holes (_interior rings_), also
+/// represented by `LineString`s.
+///
+/// Refer to [geo_types::Polygon] for information about semantics and validity.
+pub trait PolygonTrait: Sized {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying ring, which implements [LineStringTrait]
+    type RingType<'a>: 'a + LineStringTrait<T = Self::T>
+    where
+        Self: 'a;
+
+    /// The dimension of this geometry
+    fn dim(&self) -> Dimensions;
+
+    /// The exterior ring of the polygon
+    fn exterior(&self) -> Option<Self::RingType<'_>>;
+
+    /// An iterator of the interior rings of this Polygon
+    fn interiors(&self) -> impl Iterator<Item = Self::RingType<'_>> {
```

Comment:
Non-blocking: Since we're requiring them to implement `num_interiors`, should we make this `ExactSizeIterator`?

```suggestion
    fn interiors(&self) -> impl ExactSizeIterator<Item = Self::RingType<'_>> {
```

Reply:
I changed this to `DoubleEndedIterator + ExactSizeIterator` in  [`0fb5ab8` (#1157)](https://github.com/georust/geo/pull/1157/commits/0fb5ab8456f849bb089f0e0beba033244907db46)

Reply:
A follow-up thought that we could consider removing the separate `num_` methods since one can just do `interiors().len()`. But extremely minor and I don't feel strongly in any particular direction

Reply:
The implement of the iterator relies on the `num_*` method